### DISCUSSION TITLE
Bazel: Unresolved reference

### DISCUSSION THREAD

**Discussion by JishnuGoyal** [oppia/oppia-android#4367] [Q&A (Other)] [answered]
I'm  working on #4006 .
It says TestPlatformParameterModule is unresolved. Now, this module is only being used in 3 tests -- OptionsFragmentTest (it has two copies, one in app.options package and other in app.testing.options package, both are being used) ; and AdministratorControlsActivityTest.

This is the error that the CI gives: https://github.com/oppia/oppia-android/runs/6343605385?check_suite_focus=true

For these tests, there isn't any build.bazel file in the respective package that i could edit.
And for the top level Build.bazel files, I'm not able to find where to add the dependencies.

**Comment by JishnuGoyal** [comment]
I'm facing a similar problem in #4002

**Comment by BenHenning** [comment, marked as answer]
Hi @JishnuGoyal.

For tests that don't have a dedicated BUILD.bazel file in their immediate directory, you'll need to update their shared ``TEST_DEPS`` list located in the top-level BUILD.bazel file of their module (e.g. 'app'). For example, since all three of the tests you listed don't have their own test targets defined, new dependencies need to be added here: https://github.com/oppia/oppia-android/blob/7d31a49eef8de2d3f39586016d263d726b6b1d23/app/BUILD.bazel#L852.

**Reply by MohitGupta121** [reply]
Hey @JishnuGoyal if your issue is resolved please mark this discussion is resolved (marking as answered that comment) so that it's helpful to others in the community.
